Docs 9548 editorial review messages esb (#105)
mulesoft/docs-partner-manager: modules/ROOT/pages/partner-manager-create-message-type.adoc

> == Create and Configure Message Types

Reviewer: I wonder if there s/b separate tasks for Create and Configure....

Author: I agree, but this is out of scope. I have created a new story for this - https://www.mulesoft.org/jira/browse/DOCS-9847

* Source at host
* Send to partners

Message types are always managed in the host profile page, regardless of whether they are used for a single partner or for multiple trading partners. Additionally, message types are reusable across message flows for different trading partners.

> You can create message types from the Host page or from the Message Flow wizard. When you create message types during message flow configuration, Partner Manager automatically designates the usage for the message type.

Reviewer: GLOBALLY reconcile/distinguish among the terms host, host page, Host page, host profile page....

Reviewer: Ditto comments about automatic designation vs user configuration; GLOBAL
GLOBALLY clarify whether "usage" actually means specifying incoming vs outgoing, or message "type".

Author: Done

== Create a Message Type

. In Partner Manager left navigation, select *<host-name>*.
. From the *Message types* section host page:
.. Click *New*.

> ..  In the *Usage* drop-down list, select the message type:

Reviewer: Why are these substeps?

Author: Fixed

* <<receive-and-send-to-partners,*Receive from Partners*>>
* <<target-at-host,*Target at host*>>
* <<source-at-host,*Source at host*>>
* <<receive-and-send-to-partners,*Send to partners*>>
+

> If you create a message type from the message flow, Partner Manager automatically determines the *Usage* value.

Reviewer: GLOBALLY clarify this.

Author: Done

. Click *Save*. +
The message type appears in the *Message types* section on the host page.

[[receive-and-send-to-partners]]

> == Receive From and Send To Partners

Reviewer: Is this part of Configure?
Hmm...I think not create or configure, which are tasks, but a SEPARATE conceptual topic, Message Type Descriptions.

Author: Updated new story for this change

> Inbound transactions that are received from your trading partners or outbound transactions that your organization sends to your trading partners typically follow EDI standards such as X12.

Reviewer: Do we really need to define "inbound" and "outbound"? Seems wordy to me:
```suggestion
Inbound and outbound transactions between your organization and your trading partners typically follow EDI standards such as X12.
```

Use the following settings to define X12 as an inbound or outbound message type:

[%header%autowidth.spread]
|===
|Setting |Description |Default | Required

> |Format | The format of the message. |`X12`|Yes

Reviewer: If possible, standardize each Description entry to be a fragment, with no period.
The article you added is not really nec. in a table, but it doesn't much matter.

Author: Done

> |Version | Select a version from the drop-down list. Valid values include: `3010`, `3020`, `3030`, `3040`, `3050`, `3060`, `3070`, `4010`, `4020`, `4030`, `4040`, `4050`, `4060`, `5010`, `5020`, `5030`, `5040`, `5050`, `6020`, `6030`, `6040`, `6050`, `7010`, `7020`, `7030`, `7040`, `7050`

Reviewer: Unnecessary. Description here should be "List of valid versions".

Author: Done

 | None |Yes

> |Message Type |The X12 transaction set number, such as `850` for Purchase Order or `214` for Shipment Status. |None | Yes

Reviewer: If Purchase Order and Shipment Status are UI labels, then they should be in bold.

Author: Done

|Schema a| The EDI schema for the message type. Options include:

* *Use standard X12-<version>* +
Applies the standard X12 schema for the version and message type configured.
* *Customize* +

> Allows you to import a customized xref:connectors::x12-edi/x12-edi-schema-language-reference.adoc[EDI schema language] file, if your requirement is to make adjustments to the standard X12 schema.

Reviewer: GLOBALLY replace "allows" with "enables" when referring to what a user can do:
```suggestion
Enables you to import a customized xref:connectors::x12-edi/x12-edi-schema-language-reference.adoc[EDI schema language] file, if your requirement is to make adjustments to the standard X12 schema.
```

[[target-at-host]]
== Target at Host

> The application message structure (JSON or XML) to which inbound EDI transactions are translated and sent to your back-end applications.

Reviewer: Reword to be a complete sentence and to introduce the table.

Author: Done

[%header%autowidth.spread]

> |===

Reviewer: ditto re. standardizing to all fragments, without periods (or all sentences, with periods) if possible
GLOBAL for Description columns in tables

Author: Done

|Setting |Description |Default | Required
|Format a|The selected application message structure format:

* *XML*
* *JSON* |None |Yes
|Schema |The imported XSD or JSON schema file for the message type. |None |Yes

> |Name |Partner Manager automatically generates a message type name from the schema file name. This value can be modified. |None |Yes

Reviewer: ```suggestion
|Name |The message type name that Partner Manager automatically generates from the schema file name, which you can later modify |None |Yes
```
Does this description hold the answer to my question about what PM automatically "designates" as "usage" vs. manual configuration?

|===

[[source-at-host]]
== Source at Host

> Application message structures (JSON or XML) from your back-end applications that get translated into  outbound EDI transactions and sent to your trading partners.

Reviewer: ditto re. complete sentence and intro.
omit extra space as well

Author: Done

[%header%autowidth.spread]
|===
|Setting |Description |Default | Required
|Format a|The selected application message structure format:

* *XML*
* *JSON* |None |Yes
|Schema |The imported XSD or JSON schema file for the message type. |None |Yes

> |Message type identifier a|Partner Manager automatically generates a message type identifier from the schema file name. This value can be modified.

Reviewer: ditto prev edits

Author: Done

> The message identifier value is used in the dynamic message routing for outbound B2B transactions.

Reviewer: reword to avoid passive?

Author: Done

Author: Done

The input to the DataWeave map is the payload that follows the schema uploaded in the message type definition. The output must contain the following fields:

* `partnerReferenceId` +
Required identifier that uniquely identifies the receiving partner.
* `hostReferenceId` +
Optional, except when backend systems are sending outbound transactions on behalf of different internal business units that require different mapping or sender identifiers to be set on the EDI data.
* `businessDocumentKey` +

> Optional key reference value, such as `Order Number`, `Invoice Number`, and so on.

Reviewer: ```suggestion
Optional key reference value, such as `Order Number`or `Invoice Number`.
```
We want to avoid the vagueness of "and so on."
